controllers/release/adapter_test.go

```diff
@@ -935,6 +872,170 @@ var _ = Describe("Release Adapter", Ordered, func() {
+		})
+
+		It("registers the Release deployment data", func() {
+			Expect(adapter.registerDeploymentData(snapshotEnvironmentBinding, releasePlanAdmission)).To(Succeed())
+			Expect(adapter.release.Status.Deployment.Environment).To(Equal(fmt.Sprintf("%s%c%s",
+				releasePlanAdmission.Namespace, types.Separator, releasePlanAdmission.Spec.Environment)))
+			Expect(adapter.release.Status.Deployment.SnapshotEnvironmentBinding).To(Equal(fmt.Sprintf("%s%c%s",
+				snapshotEnvironmentBinding.Namespace, types.Separator, snapshotEnvironmentBinding.Name)))
+			Expect(adapter.release.IsDeploying()).To(BeTrue())
+		})
+
+		It("should not register the environment if the ReleasePlanAdmission does not reference any", func() {
+			newReleasePlanAdmission := releasePlanAdmission.DeepCopy()
+			newReleasePlanAdmission.Spec.Environment = ""
+
+			Expect(adapter.registerDeploymentData(snapshotEnvironmentBinding, newReleasePlanAdmission)).To(Succeed())
+			Expect(adapter.release.Status.Deployment.Environment).To(Equal(""))
+			Expect(adapter.release.Status.Deployment.SnapshotEnvironmentBinding).To(Equal(fmt.Sprintf("%s%c%s",
+				snapshotEnvironmentBinding.Namespace, types.Separator, snapshotEnvironmentBinding.Name)))
+			Expect(adapter.release.IsDeploying()).To(BeTrue())
+		})
+	})
+
+	Context("When registerDeploymentStatus is called", func() {
+		var adapter *Adapter
+
+		AfterEach(func() {
+			_ = adapter.client.Delete(ctx, adapter.release)
+		})
+
+		BeforeEach(func() {
+			adapter = createReleaseAndAdapter()
+		})
+
+		It("does nothing if there is no binding", func() {
+			Expect(adapter.registerDeploymentStatus(nil)).To(Succeed())
+			Expect(adapter.release.IsDeploying()).To(BeFalse())
+		})
+
+		It("does nothing if the binding doesn't have the expected condition", func() {
+			Expect(adapter.registerDeploymentStatus(snapshotEnvironmentBinding)).To(Succeed())
+			Expect(adapter.release.IsDeploying()).To(BeFalse())
+		})
+
+		It("registers the deployment status when the binding has been deployed", func() {
+			adapter.release.MarkDeploying("")
+			newSnapshotEnvironmentBinding := snapshotEnvironmentBinding.DeepCopy()
+			newSnapshotEnvironmentBinding.Status.ComponentDeploymentConditions = []metav1.Condition{
+				{
+					Type:   applicationapiv1alpha1.ComponentDeploymentConditionAllComponentsDeployed,
+					Status: metav1.ConditionTrue,
+					Reason: "Deployed",
+				},
+			}
+			Expect(adapter.registerDeploymentStatus(newSnapshotEnvironmentBinding)).To(Succeed())
+			Expect(adapter.release.HasDeploymentFinished()).To(BeTrue())
+		})
+	})
+
+	Context("When registerProcessingData is called", func() {
+		var adapter *Adapter
+
+		AfterEach(func() {
+			_ = adapter.client.Delete(ctx, adapter.release)
+		})
+
+		BeforeEach(func() {
+			adapter = createReleaseAndAdapter()
+		})
+
+		It("does nothing if there is no PipelineRun", func() {
+			Expect(adapter.registerProcessingData(nil, nil)).To(Succeed())
```

Comment:
should be passing `nil, releaseStrategy`

Reply:
Yeah, that's better actually.